web3/main.py

```diff
@@ -343,6 +344,12 @@ def pm(self) -> "PM":
             )
 
     def enable_unstable_package_management_api(self) -> None:
+        warnings.warn(
```

Comment:
Tiny nit, take or leave, but we can move this down to where we actually attach the PM module, inside of the if block. 

Unrelated, but it would be nicer to move the local import in there too if we go that route.

```py
def enable_unstable_package_management_api(self) -> None:
    if not hasattr(self, "_pm"):
        # warning here
    
        from web3.pm import PM  # noqa: F811
        self.attach_modules({"_pm": PM})
```

Reply:
moved!